### ISSUE TITLE
Prompt Processing slower post 1.50?

### ISSUE THREAD

**Issue by heyyyyitsmelodyyy** [LostRuins/koboldcpp#587]
I've tested the performance of versions 1.50 and version 1.51-.53 (using the same settings/parameters, regenerating responses for the same message, etc.) and found that that 1.50 and before process prompts noticeably faster, on an order of magnitude of 80s-100s. Token generation is more or less the same. Has anyone else experienced anything similar?

GUI Settings:
![ZVnhiDq](https://github.com/LostRuins/koboldcpp/assets/90802481/4c7d3db7-3c56-4781-bc8f-b52709a3ef7d)

**Comment by LostRuins** [comment]
Can you see if toggling MMQ and high priority off make a difference? 

If not, try reduce number of layers by 2 and reduce thread count by 2, and see if that is faster. 

(Because on some drivers when VRAM runs out they switch to shared memory, and on some devices too many threads actually reduce performance)

**Comment by heyyyyitsmelodyyy** [comment]
Disabling 'Use mmq' and 'High Priority' led to the prompt processing going to around 10 minutes before some error pops up and crashes koboldcpp. I wasn't quick enough to see what the error said.

I performed multiples tests where the layers and threads were reduced 2. The numbers have varied a lot, but generally shows that the reducing by 2 slows down prompt processing noticeably or not much change, with one test going from 283s -> 554s. Another went from 279s -> 293s. So, either significantly slower or barely any difference at all, but not faster in any test.

I did a single test on koboldcpp 1.50 which had 46/7 layers/threads at 111s and 44/5 layers/threads at 123s. Not much change here either.

Additional info:
20b model
GPU: Nvidia 4080 RTX
CPU: Ryzen 7 5800X3D 
RAM: 64gb DDR4

**Comment by LostRuins** [comment]
What model are you trying to load?

**Comment by heyyyyitsmelodyyy** [comment]
Noromaid-20B. Mostly tested with Q5_K_M, but produced similar results with Q5_K_S as well.

**Comment by LostRuins** [comment]
Hmm I am not sure, looking through the code nothing significant has changed between 1.50 and 1.51 that could cause it. How long is your prompt? If you enable mmap does it help?

**Comment by VL4DST3R** [comment]
Could this be caused by 1.52's partial per-layer KV offloading? The whole "`--usecublas lowvram`  which trades ingest speed with generation speed" discussion?

**Comment by LostRuins** [comment]
It's possible, but then using lowvram with the original layer count should give identical results again.

**Comment by DistantThunder** [comment]
Hi, I don't know if this is related but using ROCm 5.7 here, partial offloading to GPU also tends to be much slower the more layers you offload, which sounds counter intuitive:

* Try to use: https://huggingface.co/TheBloke/Mixtral-8x7B-Instruct-v0.1-GGUF/tree/main (Q4)
* KoboldCPP 1.54.yr0-ROCm

* [slow_koboldcpp.log](https://github.com/LostRuins/koboldcpp/files/13813363/slow_koboldcpp.log)
  * > llm_load_tensors: offloaded 15/33 layers to GPU
    > [...]
    >ContextLimit: 33/1600, Processing:31.91s (1994.4ms/T), Generation:38.56s (2268.5ms/T), Total:70.48s (0.24T/s)

 vs 
>llm_load_tensors: system memory used  = 9564.37 MiB
llm_load_tensors: VRAM used           = 15651.88 MiB
llm_load_tensors: offloading 20 repeating layers to GPU
llm_load_tensors: offloaded 20/33 layers to GPU
>
>[...] 
Processing Prompt (16 / 16 tokens)
Generating (15 / 120 tokens)
(EOS token triggered!)
ContextLimit: 31/1600, Processing:72.88s (4555.1ms/T), Generation:110.21s (7347.1ms/T), Total:183.09s (0.08T/s)

**Comment by Drael64** [comment]
1.51. Yeah, I have the same bug, just logged it.

**Comment by Skydea777** [comment]
Yeah same Issue I also posted here: "https://github.com/LostRuins/koboldcpp/issues/619" and I am losing my mind with this Problem!

Interesstingly it does not happens when I compile my own Cuda Version of LLamaCpp (Using the latest build from master and Cuda Toolkit 12.2.2 and Visual Studio 2019 Community Edition) with CMake. 
the build was executet via:
"cmake .. -DLLAMA_CUBLAS=ON"
"cmake --build . --config Release"
(I reinstalled VS 2019 Community Edition and Cuda Toolkit. In that order and set all needed Enviroment Variables.)
Everything works wonderful with identical settings used in KoboldCPP while using the "Main.exe" This test withoug GPU Layers:
.\main.exe -i --color -m "C:\AI\CPP\Models-CPP\Noromaid-13B-0.4-DPO.q5_k_m.gguf" -r "Tell me about yourself."
![grafik](https://github.com/LostRuins/koboldcpp/assets/156956074/503a49e2-0ba2-46ef-8ae0-8ace31bc02d7)
Also with GPU Layers (this above was just a test for my sanity):
.\main.exe -i --color -m "C:\AI\CPP\Models-CPP\Noromaid-13B-0.4-DPO.q5_k_m.gguf" -ngl 26 -r "Tell me about yourself."
![grafik](https://github.com/LostRuins/koboldcpp/assets/156956074/7a8fea4f-f47c-4aba-9ae3-d5ec290db89b)

Using the "server.exe" in combination with Sillytaven (Regardles of Settings in there) works pretty well.
.\server -v -t 8 -tb 8 -c 8192 -b 512 -ngl 26 -m "C:\AI\CPP\Models-CPP\Noromaid-13B-0.4-DPO.q5_k_m.gguf"
Speed is not really impaired even if it is hallucinating heavily... for what ever reason.


Using the Prebuild Version von LLama.cpp Version b1961
I get pretty much the exact same results:
![grafik](https://github.com/LostRuins/koboldcpp/assets/156956074/73cd3f85-e90d-49b2-856e-a4882a2b62a3)

Also Server.exe still halluciating like crazy.. i mean .. rolling my chair back then squeezing my ankle  and after that saying... gibberish with %COSTCO ... interessting.
But it seems to work.


KoboldCpp 1.54:
Settings
![grafik](https://github.com/LostRuins/koboldcpp/assets/156956074/7719bee5-c803-48ee-9757-e6397aef7b16)

Result:
![grafik](https://github.com/LostRuins/koboldcpp/assets/156956074/cdf66d2d-a0ad-4b12-8406-2b1539616e63)


KoboldCpp 1.55.1:
![grafik](https://github.com/LostRuins/koboldcpp/assets/156956074/b45ffd79-59c0-47e0-a331-053754223c02)


Result:
![grafik](https://github.com/LostRuins/koboldcpp/assets/156956074/239646e9-5ab9-4fc5-b435-b570d05ad894)



As you can see, the Bug is really only happening in 1.55+ of Kobold Cpp, it is completly irrelevant how much Layers I choose to use, if I use LowVram or not, it ... just stopped working and I am out of ideas, no seriously, I am searching for days why this is not working! 

Please Help! :(
